flow/raster_cache.cc

```diff
@@ -118,51 +127,63 @@ static RasterCacheResult Rasterize(
     DrawCheckerboard(canvas, logical_rect);
   }
 
-  return {surface->makeImageSnapshot(), logical_rect};
+  return std::make_unique<SkRasterCacheResult>(surface->makeImageSnapshot(),
+                                               logical_rect);
 }
 
-RasterCacheResult RasterizePicture(SkPicture* picture,
-                                   GrContext* context,
-                                   const SkMatrix& ctm,
-                                   SkColorSpace* dst_color_space,
-                                   bool checkerboard) {
+SkRasterCacheImageDelegate SkRasterCacheImageDelegate::instance;
+
+std::unique_ptr<RasterCacheResult> SkRasterCacheImageDelegate::RasterizePicture(
+    SkPicture* picture,
+    GrContext* context,
+    const SkMatrix& ctm,
+    SkColorSpace* dst_color_space,
+    bool checkerboard) const {
   return Rasterize(context, ctm, dst_color_space, checkerboard,
                    picture->cullRect(),
                    [=](SkCanvas* canvas) { canvas->drawPicture(picture); });
 }
 
+std::unique_ptr<RasterCacheResult> SkRasterCacheImageDelegate::RasterizeLayer(
+    PrerollContext* context,
+    Layer* layer,
+    const SkMatrix& ctm,
+    bool checkerboard) const {
+  return Rasterize(
+      context->gr_context, ctm, context->dst_color_space, checkerboard,
+      layer->paint_bounds(), [layer, context](SkCanvas* canvas) {
+        SkISize canvas_size = canvas->getBaseLayerSize();
+        SkNWayCanvas internal_nodes_canvas(canvas_size.width(),
+                                           canvas_size.height());
+        internal_nodes_canvas.addCanvas(canvas);
+        Layer::PaintContext paintContext = {
+            (SkCanvas*)&internal_nodes_canvas,
```

Comment:
nit: argument name comments would be a bit helpful here.  Google has standardized on `/*arg_name=*/

Reply:
Hmmm, I find that notation ugly and hard to read in general. It's only appeared in a couple of places mostly dealing with Fuchsia so far and the few that I looked at look like what was once simple code becoming text soup. That discussion is probably for another venue, though.

In this particular case, every field is being copied over except for a couple. The somewhat clean code that makes it obvious that this particular process is a "mostly copy transfer/constructor" would start to look like `/* really_long_and_obvious_name= */ really_long_and_obvious_name,`. Is that adding any value at that point?

The only two lines which are less than obvious are the null for the view_embedder (which is basically, we don't get to this code if there was a view_embedder in the first place so I could replace it with `context->view_embedder` and it would be identical and more obvious, but waste code) and passing the supplied `canvas` variable to the `leaf_nodes_canvas` field only because the field has an adorned name.

I didn't write this code, I moved it a few lines in the source file. I feel it's pretty obvious so I don't see much value in adding what amounts to obfuscation notations unrelated to the actual fix that is happening just because a diff engine's robotic algorithm makes it look like new code in the review tool.

Reply:
@gaaclarke I've made some updates on the code that you commented on, did they satisfy the changes you were requesting?

Reply:
Not my rule, google's: https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

Reply:
From that page:

```
When the meaning of a function argument is nonobvious, consider one of the following remedies:
- ...
- ...
- ...
- ...
- As a last resort, use comments to clarify argument meanings at the call site.
```

I don't think the remaining arguments qualify as "when the meaning is nonobvious" and I don't think this situation calls for a last resort. Further, I think that blindly using that last resort rule on every function call goes against everything that paragraph stands for. Their own example of a clean source example only uses it for one argument and that argument isn't clear from context.

The format isn't dictated, they only indicate that a comment be used - and I used a comment in the manner that all of the surrounding code in that modules uses.